app/src/main/java/org/mozilla/fenix/tabtray/TabTrayViewHolder.kt

```diff
@@ -94,49 +98,98 @@ class TabTrayViewHolder(
 
         // Media state
         playPauseButtonView.increaseTapArea(PLAY_PAUSE_BUTTON_EXTRA_DPS)
-        with(playPauseButtonView) {
-            invalidate()
-            Do exhaustive when (store.state.getMediaStateForSession(tab.id)) {
-                MediaState.State.PAUSED -> {
-                    showAndEnable()
-                    contentDescription =
-                        context.getString(R.string.mozac_feature_media_notification_action_play)
-                    setImageDrawable(
-                        AppCompatResources.getDrawable(context, R.drawable.media_state_play)
-                    )
-                }
 
-                MediaState.State.PLAYING -> {
-                    showAndEnable()
-                    contentDescription =
-                        context.getString(R.string.mozac_feature_media_notification_action_pause)
-                    setImageDrawable(
-                        AppCompatResources.getDrawable(context, R.drawable.media_state_pause)
-                    )
+        if (Config.channel.isNightlyOrDebug) {
+            with(playPauseButtonView) {
+                invalidate()
+                val sessionState = store.state.findTabOrCustomTab(tab.id)
+                when (sessionState?.mediaSessionState?.playbackState) {
+                    MediaSession.PlaybackState.PAUSED -> {
+                        showAndEnable()
+                        contentDescription =
+                            context.getString(R.string.mozac_feature_media_notification_action_play)
+                        setImageDrawable(
+                            AppCompatResources.getDrawable(context, R.drawable.media_state_play)
+                        )
+                    }
+
+                    MediaSession.PlaybackState.PLAYING -> {
+                        showAndEnable()
+                        contentDescription =
+                            context.getString(R.string.mozac_feature_media_notification_action_pause)
+                        setImageDrawable(
+                            AppCompatResources.getDrawable(context, R.drawable.media_state_pause)
+                        )
+                    }
+
+                    else -> {
+                        removeTouchDelegate()
+                        removeAndDisable()
+                    }
```

Comment:
Oh, ugh, I didn't think of how much work it would be to separate the two implementations in the UI layers.

Reply:
Yeah :).  But once we're confident with the new MediaSession API changes, I can remove all of this and use a much cleaner implementation.